src/lib/npm.js

```diff
@@ -42,13 +42,27 @@ async function getNpmToken({npm, options}) {
   log.info('Successfully created npm token.');
 }
 
+function getRegistry(pkg, conf) {
+  if (pkg.publishConfig && pkg.publishConfig.registry) return pkg.publishConfig.registry
+    
+  if (pkg.name[0] !== '@') return conf.get('registry') || 'https://registry.npmjs.org/'
+    
+  const scope = pkg.name.split('/')[0]
+  const scopedRegistry = conf.get(`${scope}/registry`)
+
+  if (scopedRegistry) return scopedRegistry
+
+  return conf.get('registry') || 'https://registry.npmjs.org/'
+
+}
```

Comment:
code looks good, but moving forward we really need to improve the test / code coverage situation :)

Reply:
Agreed